src/tmle_inputs/from_actors.jl

```diff
@@ -12,6 +12,24 @@ function treatments_from_actors(bqtl_file, env_file, trans_actors_prefix)
 end
 
 
+tf_error() = throw(ArgumentError("No TF specified to filter SNPs on"))
+missing_snps_error() = throw(ArgumentError("No SNPs provided to filter on"))
+
+filter_snps_by_tf(df::Nothing, tf_name::Nothing) = nothing
+filter_snps_by_tf(df::DataFrame, tf_name::Nothing) = tf_error()
+filter_snps_by_tf(df::Nothing, tf_name::AbstractString) = missing_snps_error()
+function filter_snps_by_tf(df::DataFrame, tf_name::AbstractString)
+    return filter(row -> row.TF == tf_name, df)
+end
+
+function filter_bqtls_by_tf(bqtl_df::DataFrame, tf_name::AbstractString)
+    return filter_snps_by_tf(bqtl_df, tf_name)
+end
+
+function filter_transactors_by_tf(trans_actors::Union{Vector{DataFrame}}, tf_name::AbstractString)
```

Comment:
Will this require passing in the bqtls variable as a vector? ie. when you pass in the bqtls variable into the parameters_from_actors() function, will you need to transform it to [bqtls]? Because currently it is just a DataFrame.

Reply:
addressed: e4c2e131323f5032ff8a3be17776638ec0b1ba05